### DISCUSSION TITLE
Suggestion - Node Inspector Copy to Clipboard, Paste Code Snippet

### DISCUSSION THREAD

**Discussion by brokenstyli** [motion-canvas/motion-canvas#1042] [Ideas]
Hi, I have suggestion for improving MotionCanvas's Node Inspector, specifically the copy-to-clipboard function it has currently.

I'm not entirely sure if this is possible, but if it is... it would very much help a beginner user like myself.

Currently if you click on an object in the Scene Preview and open up the Node Inspector, you can click on a variable value and it copies the value to the clipboard. I think a novel use for copy-to-clipboard would be if you clicked on a variable name/value, it gives you the exact code snippet, with both the value already in it and the most up-to-date syntax for MotionCanvas.

For example, (in v3.15.2) suppose you already have a Circle in the scene with `ref={myCircle}` at position `x={0}, y={15}` and want to translate the Circle 10 pixels down over one second.

* click on the Circle in the preview to open up the Node Inspector
* click the arrow in "position" to split/reveal the Y position
* click the y position value
* the following is copied to clipboard;
`myCircle.position.y(15),`

This would give me the code with the right syntax, and I could start modifying it immediately and append the required `.to(-10,1)` to animate it downwards.

And if I were to need to animate rotation, scale, etc... with some forethought, I can prep an entire `yield* all();` code block ahead of time by repeatedly copy-pasting from the Node Inspector, and then make my adjustments as needed (similar to After Effects where if I know I need three separate Sliders for expressions to control position, scale, rotation of a layer, I can just drag three Slider effects onto a layer ahead of time and then adjust their functionality when I get around to it).

To my knowledge, the Node Inspector already does something similar? If you just click the "position" value without expanding the arrow, the text it copies is `{"x":0,"y":15}`, so it looks like it's already distinguishing between different variables and pre-pend/appending strings. Wouldn't it just be the same thing, just pre-pending a different string like `myCircle.position.` ?

If this is possible, then obviously it should also come with a toggle in the settings tab so if you only want/need the raw values you can switch back to it.

**Comment by aarthificial** [comment]
While we could add an ability to copy the property path (like `position.x(5)`) there is no way for us to know the name of the variable used to store the reference (i.e `myCircle`)
